components/logins/src/db.rs

```diff
@@ -419,6 +419,82 @@ impl LoginDb {
         Ok(login)
     }
 
+    pub fn import_multiple(&self, logins: &[Login]) -> Result<u64> {
+        let tx = self.unchecked_transaction()?;
+        let now_ms = util::system_time_ms_i64(SystemTime::now());
+        let sql = format!(
+            "INSERT OR IGNORE INTO loginsL (
+                hostname,
+                httpRealm,
+                formSubmitURL,
+                usernameField,
+                passwordField,
+                timesUsed,
+                username,
+                password,
+                guid,
+                timeCreated,
+                timeLastUsed,
+                timePasswordChanged,
+                local_modified,
+                is_deleted,
+                sync_status
+            ) VALUES (
+                :hostname,
+                :http_realm,
+                :form_submit_url,
+                :username_field,
+                :password_field,
+                :times_used,
+                :username,
+                :password,
+                :guid,
+                :time_created,
+                :time_last_used,
+                :time_password_changed,
+                :local_modified,
+                0, -- is_deleted
+                {new} -- sync_status
+            )",
+            new = SyncStatus::New as u8
+        );
+        let mut num_failed = 0;
+        for login in logins {
+            if login.check_valid().is_err() {
+                log::warn!("Skipping login {} as it is invalid.", login.guid);
+                continue;
+            }
+            let old_guid = &login.guid; // Keep the old GUID around so we can debug errors easily.
+            let guid = Guid::random(); // Override any provided GUID.
+            match self.execute_named_cached(
+                &sql,
+                named_params! {
+                    ":hostname": login.hostname,
+                    ":http_realm": login.http_realm,
+                    ":form_submit_url": login.form_submit_url,
+                    ":username_field": login.username_field,
+                    ":password_field": login.password_field,
+                    ":username": login.username,
+                    ":password": login.password,
+                    ":guid": guid,
+                    ":time_created": login.time_created,
+                    ":times_used": login.times_used,
+                    ":time_last_used": login.time_last_used,
+                    ":time_password_changed": login.time_password_changed,
+                    ":local_modified": now_ms,
```

Comment:
Not sure this is what we want. The logins that are imported could be old and out of date. This will make us prefer them to newer records from the server.

Reply:
Actually I guess we aren't doing this for sync users? In which this is fine?

Reply:
Yup only for non-sync users.